### DISCUSSION TITLE
Third-party/custom rules?

### DISCUSSION THREAD

**Discussion by osdiab** [rome/tools#3852] [Q&A]
Hi all, I can't find any mention of this in the docs or discussions/issues about it so I'm not sure if I'm just missing something but, does Rome support or have any plans to support writing custom or third party lint rules? Thanks!

**Comment by MichaReiser** [comment]
We want to explore Plugin support eventually but isn't on our immediate roadmap. What's your need for custom rules? Is it because Rome doesn't support a popular rule today, or do you want to create a project-specific rule?

**Reply by osdiab** [reply]
In our code we create a lot of project-specific rules to guide the team on what we should and shouldn't use - for instance wrapping a library with our own code and then banning direct imports from that library to direct people to use our wrapper; or blocking certain patterns, types, or imports in libraries that we consider code smells. That's one common one; another is just that there's a lot of library-specific ESLint plugins in the wild (e.g. Emotion's ESLint plugin), and so having to sacrifice those if we wanted to switch to Rome is a bummer (and it seems like a weird situation to be in where Rome would need to subsume everything every plugin in the ESLint ecosystem does to be legitimately complete).
